metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -391,41 +393,60 @@ public void replay(UnregisterBrokerRecord record) {
     }
 
     public void replay(FenceBrokerRecord record) {
-        int brokerId = record.id();
-        BrokerRegistration registration = brokerRegistrations.get(brokerId);
-        if (registration == null) {
-            throw new RuntimeException(String.format("Unable to replay %s: no broker " +
-                "registration found for that id", record.toString()));
-        } else if (registration.epoch() !=  record.epoch()) {
-            throw new RuntimeException(String.format("Unable to replay %s: no broker " +
-                "registration with that epoch found", record.toString()));
-        } else {
-            if (heartbeatManager != null) heartbeatManager.register(brokerId, true);
-            brokerRegistrations.put(brokerId, registration.cloneWithFencing(true));
-            updateMetrics(registration, brokerRegistrations.get(brokerId));
-            log.info("Fenced broker: {}", record);
-        }
+        replayRegistrationChange(record, record.id(), record.epoch(), Optional.of(true));
     }
 
     public void replay(UnfenceBrokerRecord record) {
-        int brokerId = record.id();
-        BrokerRegistration registration = brokerRegistrations.get(brokerId);
-        if (registration == null) {
+        replayRegistrationChange(record, record.id(), record.epoch(), Optional.of(false));
+    }
+
+    public void replay(BrokerRegistrationChangeRecord record) {
+        replayRegistrationChange(record, record.brokerId(), record.brokerEpoch(), getFencingChange(record.fenced()));
+    }
+
+    private Optional<Boolean> getFencingChange(byte value) {
+        switch (value) {
+            case -1:
+                return Optional.of(false);
+            case 0:
+                return Optional.empty();
+            case 1:
+                return Optional.of(true);
```

Comment:
Can define and create Java symbols for this constants? E.g. a Java enum. Don't we need to use this constants in the broker's metadata listener and delta?